earthpy/plot.py

```diff
@@ -497,8 +501,14 @@ def hist(
                 the_color = colors[0]
             else:
                 the_color = colors[i]
+            if not hist_range:
+                hist_range = [np.nanmin(band), np.nanmax(band)]
```

Comment:
this should be a tuple not a list `range = (np.nanmin(arr), np.nanmax(arr))` and also should use the input array. you only have to calculate range once for the entire array regardless of single vs multiple band. if there are many bands, please calculate the range on the ENTIRE ARRAY.  so it's not in the loop